src/System.Buffers.Primitives/System/Buffers/OwnedBuffer.cs

```diff
@@ -13,45 +12,17 @@ public abstract class OwnedBuffer<T> : IDisposable, IRetainable
 
         public abstract int Length { get; }
 
-        public abstract Span<T> Span { get; }
+        public abstract Span<T> AsSpan(int index, int length);
```

Comment:
If you buffer is in memory, then you are right. But OwnedBuffer is an abstraction and not all it's data have to be in memory.